examples/drivers/windows/otLwf/tunnel.c

```diff
@@ -0,0 +1,1745 @@
+        break;
+
+    case SPINEL_STATUS_CCA_FAILURE:
+        ret = kThreadError_ChannelAccessFailure;
+        break;
+
+    case SPINEL_STATUS_ALREADY:
+        ret = kThreadError_Already;
+        break;
+
+    case SPINEL_STATUS_ITEM_NOT_FOUND:
+        ret = kThreadError_NotFound;
+        break;
+
+    default:
+        if (error >= SPINEL_STATUS_STACK_NATIVE__BEGIN && error <= SPINEL_STATUS_STACK_NATIVE__END)
+        {
+            ret = (ThreadError)(error - SPINEL_STATUS_STACK_NATIVE__BEGIN);
+        }
+        else
+        {
+            ret = kThreadError_Failed;
+        }
+        break;
+    }
+
+    return ret;
+}
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+NDIS_STATUS 
+otLwfInitializeTunnelMode(
+    _In_ PMS_FILTER pFilter
+    )
+{
+    NDIS_STATUS Status = NDIS_STATUS_SUCCESS;
+    uint32_t InterfaceType = 0;
+    HANDLE threadHandle = NULL;
+
+    LogFuncEntry(DRIVER_DEFAULT);
+
+    pFilter->tunTIDsInUse = 0;
+    pFilter->tunNextTID = 1;
+    
+    NdisAllocateSpinLock(&pFilter->tunCommandLock);
+    InitializeListHead(&pFilter->tunCommandHandlers);
+    
+    KeInitializeEvent(
+        &pFilter->TunWorkerThreadStopEvent,
+        SynchronizationEvent, // auto-clearing event
+        FALSE                 // event initially non-signalled
+        );
+    KeInitializeEvent(
+        &pFilter->TunWorkerThreadAddressChangedEvent,
+        SynchronizationEvent, // auto-clearing event
+        FALSE                 // event initially non-signalled
+        );
+
+    // Start the worker thread
+    Status = PsCreateSystemThread(
+                &threadHandle,                  // ThreadHandle
+                THREAD_ALL_ACCESS,              // DesiredAccess
+                NULL,                           // ObjectAttributes
+                NULL,                           // ProcessHandle
+                NULL,                           // ClientId
+                otLwfTunWorkerThread,           // StartRoutine
+                pFilter                         // StartContext
+                );
+    if (!NT_SUCCESS(Status))
+    {
+        LogError(DRIVER_DEFAULT, "PsCreateSystemThread failed, %!STATUS!", Status);
+        goto error;
+    }
+
+    // Grab the object reference to the worker thread
+    Status = ObReferenceObjectByHandle(
+                threadHandle,
+                THREAD_ALL_ACCESS,
+                *PsThreadType,
+                KernelMode,
+                &pFilter->TunWorkerThread,
+                NULL
+                );
+    if (!NT_VERIFYMSG("ObReferenceObjectByHandle can't fail with a valid kernel handle", NT_SUCCESS(Status)))
+    {
+        LogError(DRIVER_DEFAULT, "ObReferenceObjectByHandle failed, %!STATUS!", Status);
+        KeSetEvent(&pFilter->TunWorkerThreadStopEvent, IO_NO_INCREMENT, FALSE);
+    }
+
+    // Query the interface type to make sure it is a Thread device
+    Status = otLwfGetTunProp(pFilter, NULL, SPINEL_PROP_INTERFACE_TYPE, SPINEL_DATATYPE_UINT_PACKED_S, &InterfaceType);
```

Comment:
Yes. The function is implemented below. It synchronizes sending a transaction request, waiting for the response, and unpacking the spinel command.